packages/next/server/base-server.ts

```diff
@@ -1197,7 +1197,22 @@ export default abstract class Server {
           if (result.response.headers.has('Location')) {
             const value = result.response.headers.get('Location')!
             const rel = relativizeURL(value, initUrl)
-            result.response.headers.set('Location', rel)
+
+            const { parsedDestination } = prepareDestination({
+              appendParamsToQuery: true,
+              destination: rel,
+              params: _params,
+              query: parsedUrl.query,
+            })
+            const location = formatUrl(parsedDestination)
+
+            if (location !== rel) {
+              Log.warn(
+                `Merging query params for response, from ${rel} to ${location}`
+              )
```

Comment:
Should I create a documentation page or anything related to that, that explains query params merging? Should we have this warning at all?